frontend/src/components/profileList/ListRow.vue

```diff
@@ -0,0 +1,90 @@
+<template>
+	<!-- Route to profile via avatar link -->
+	<div v-if="!userStore.isBlocked(user.id)">
+		<Avatar :avatar="user.avatar" :is-online="true" v-on:click="routeToProfile(user.id)"/>
+	</div>
+	<div v-else>
+		<Avatar :avatar="user.avatar" :is-online="false" class="grayedOut"/>
+	</div>
+
+
+	<!-- Route to profile via username link -->
+	<div v-if="!userStore.isBlocked(user.id)">
+		<a href="#" v-on:click.prevent="routeToProfile(user.id)">{{user.name}}</a>
+	</div>
+	<div v-else>
+		<a href="#" class="grayedOut">{{user.name}}</a>
+	</div>
+	
+	
+	<!-- Route to chat -->
+	<div v-if="!userStore.isBlocked(user.id)">
+		<a href="#" v-on:click.prevent="routeToChat(user.id)">Chat</a>
+	</div>
+	<div v-else>
+		<a href="#" class="grayedOut">Chat</a>
+	</div>
+
+
+	<FriendInvite :user-id="user.id"  />
+	<BlockUser :user-id="user.id"  />
+	
+</template>
+
+<script lang="ts">
+import { defineComponent, type PropType, ref } from 'vue';
+import HorizontalAvatarAndUserName from '@/components/user-info/HorizontalAvatarAndUserName.vue';
+import type { User } from '@/types/User';
+import { useUserStore } from '@/stores/userStore';
+import Avatar from '@/components/profileList/Avatar.vue'
+import FriendInvite from '@/components/profileList/FriendInvite.vue'
+import BlockUser from '@/components/profileList/BlockUser.vue'
+import router from '@/router';
+
+let isBlocked = ref(false);
+export default defineComponent({
+	name: 'ListRow',
+
+	setup() {
+		const userStore = useUserStore();
+		userStore.refreshData();
+
+		return {
+			userStore,
+		};
+	},
+
+	components: {
+		HorizontalAvatarAndUserName,
+		Avatar,
+		FriendInvite,
+		BlockUser
+	},
+
+	props: {
+		user: {
+			type: Object as PropType<User>,
+			required: true,
+		},
+		online: Boolean,
+	},
+	methods: {
+		async routeToChat(userId: number) {
+			console.log(`Starting chat with ${userId}`);
+			await router.push({ name: 'chat', params: {id: userId} })
+		},
+		async routeToProfile(userId: number) {
+			await router.push({name: 'profile', params: {id : userId}})
+		},
+	},
+});
+</script>
+
+<style>
```

Comment:
should this have a "scoped" attribute? Or does that break it since the class is not used in this component but in a sub-component?

Reply:
On second thought, and better look, this does not seem to be used (the .avatar)
so can be removed

Reply:
Indeed un-used, now removed